browser/app/profile/firefox.js

```diff
@@ -1865,8 +1865,8 @@ pref("toolkit.coverage.enabled", false, locked);
 pref("toolkit.coverage.endpoint.base", "");
 
 // Discovery prefs
-pref("browser.discovery.enabled", false);
-pref("browser.discovery.containers.enabled", true);
+pref("browser.discovery.enabled", false, locked);
+pref("browser.discovery.containers.enabled", false, locked);
```

Comment:
Why lock?

Reply:
@grahamperrin `browser.discovery.enabled` has to do with personalising recommendations.  And that is also disabled by Waterfox hard-disabling health report telemetry.  IMO locking this pref to `false` seems appropriate since it can't do anything else in Waterfox.

See related code -
https://github.com/MrAlex94/Waterfox/blob/19fe621225c69b3a8df8f60cc5d68bd0422601e4/browser/modules/Discovery.jsm
https://github.com/MrAlex94/Waterfox/blob/19fe621225c69b3a8df8f60cc5d68bd0422601e4/toolkit/mozapps/extensions/content/extensions.js#L2343-L2353